libvast/vast/concept/support/detail/is_specialization_of.hpp

```diff
@@ -0,0 +1,23 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <caf/detail/type_traits.hpp>
+
+#include <type_traits>
+
+namespace vast::detail {
+
+template <template <class...> class Template, class Instantiation>
+struct is_specialization_of : std::false_type {};
```

Comment:
I think this needs some note that this only works for templates that do not have NTTPs.

Also, can you move this to `vast/detail/is_specialization_of.hpp`?